### ISSUE TITLE
Address Field layout edits don't stick

### ISSUE THREAD

**Issue by KatieMFritz** [doublesecretagency/craft-smartmap#36]
After creating and saving a new Address field, the sub fields go into the following order:
![screen shot 2019-03-05 at 10 49 09 am](https://user-images.githubusercontent.com/7085184/53796447-7842be00-3f34-11e9-9271-950135a4b74d.png)

If I change them and save the field, upon opening the field again, they're back to that default.

Local site info:
![screen shot 2019-03-05 at 10 50 49 am](https://user-images.githubusercontent.com/7085184/53796509-9f00f480-3f34-11e9-95e5-1cba92ffce33.png)
![screen shot 2019-03-05 at 10 50 55 am](https://user-images.githubusercontent.com/7085184/53796512-9f998b00-3f34-11e9-9903-1e074a7a27c0.png)

![screen shot 2019-03-05 at 10 51 38 am](https://user-images.githubusercontent.com/7085184/53796532-ac1de380-3f34-11e9-9a5b-d8ffc898b127.png)

**Comment by lindseydiloreto** [comment]
Hmm, interesting. My suspicion is that this has something to do with the `project.yaml` file.

Can you see if the field shows up in there? And if it does, can you grab a screenshot or snippet and post it here?

**Comment by KatieMFritz** [comment]
```yaml
  840ba690-e793-40bc-8c3b-538672264018:
    contentColumnType: string
    fieldGroup: fed981b0-3708-4ebb-b4a2-d1a1ed72a61c
    handle: testAddress
    instructions: ''
    name: 'Test Address'
    searchable: true
    settings:
      dragPinDefault: ''
      dragPinLatitude: '0'
      dragPinLongitude: '0'
      dragPinZoom: '4'
      layout:
        city:
          enable: 1
          width: 50
        country:
          enable: 1
          width: 100
        lat:
          enable: 1
          width: 50
        lng:
          enable: 1
          width: 50
        state:
          enable: 1
          width: 15
        street1:
          enable: 1
          width: 100
        street2:
          enable: 1
          width: 100
        zip:
          enable: 1
          width: 35
    translationKeyFormat: null
    translationMethod: none
    type: doublesecretagency\smartmap\fields\Address
```

**Comment by lindseydiloreto** [comment]
I can't say _why_ this is happening, but I do see the issue...

The `layout` array has been rearranged into alphabetical order. That's not supposed to be the case (or at least, it didn't do that before Craft 3.1). If it's possible to rearrange that in your project config, that should hopefully solve the problem. I'll still need to get to the bottom of what caused that, and determine how widespread the issue is.

Thanks for bringing it to my attention!

**Comment by KatieMFritz** [comment]
I tried changing it in `project.yaml`. It still showed in alpha order in the CP. I rearranged it in the CP and saved, and `project.yaml` went back to alpha order. 😒

Thanks for looking into it!

**Comment by lindseydiloreto** [comment]
I've got a fix in place for this on the dev branch, give this a shot...

1. Backup your database (just in case).
2. Update the `craft-smartmap` version in your `composer.json` file, like so...

```
"doublesecretagency/craft-smartmap": "dev-v3-dev"
```

3. Run `composer update`.

 Let me know if that does the trick for you! 🙂

**Comment by KatieMFritz** [comment]
That worked! Thanks, Lindsey.

**Comment by lindseydiloreto** [comment]
Woohoo!! Thanks for confirming. 🎉 

I'm gonna leave this thread open until I officially push it live.

**Comment by lindseydiloreto** [comment]
This has been fixed in **v3.2.2**

Coincidentally, the latest version of Craft no longer alphabetizes nested arrays. 🙄 

Regardless, I'm happy that I implemented a fix on the Smart Map side of things. It's now in a much better position to be considered "future-proof".

Thanks again for reporting!
